### PR TITLE
Document that 127.0.0.2 has to be added to the loopback interface

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
While debugging with @Kostov6 on his local setup, we were not able to verify that https://github.com/gardener/gardener/pull/7526 fixes the 2nd Seed cluster creation. It was failing with:

```
% make kind2-up KIND_ENV=local

docker: Error response from daemon: Ports are not available: exposing port TCP 127.0.0.2:9443 -> 0.0.0.0: listen tcp 127.0.0.2:9443: bind can't assign requested address.
make: *** [kind2-up] Error 1
```

While on my side the 2nd Seed cluster creation was working with the fix from https://github.com/gardener/gardener/pull/7526.
We noticed that the instruction `sudo ip addr add 127.0.0.2 dev lo0` is not present in `deployment/getting_started_locally.md` while it is present in `development/getting_started_locally.md`.
We had to execute the above command to add 127.0.0.2 to `lo0`. After that he was able to successfully created the 2nd Seed cluster.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
N/A

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

File: docs/deployment/getting_started_locally.md [hunk 1]
```diff
@@ -173,8 +173,13 @@ To access the `Shoot`, you can acquire a `kubeconfig` by using the [`shoots/admi
 
 ## (Optional): Setting Up a Second Seed Cluster
 
-There are cases where you would want to create a second cluster seed in your local setup. For example, if you want to test the [control plane migration](../usage/control_plane_migration.md) feature. The following steps describe how to do that.
+There are cases where you would want to create a second seed cluster in your local setup. For example, if you want to test the [control plane migration](../usage/control_plane_migration.md) feature. The following steps describe how to do that.
 
+```bash
+sudo ip addr add 127.0.0.2 dev lo0                                     # adding 127.0.0.2 ip to the loopback interface
```
Comment (vpnachev) [inline docs/deployment/getting_started_locally.md:181]: This is applicable only to MacOS.

**Comment by ialidzhikov** [comment]
/hold
to check https://github.com/gardener/gardener/pull/7528#discussion_r1112681071

**Comment by timuthy** [comment]
As @ScheererJ pointed out, this action actually only needs to be performed on MacOS since on Linux based distros all `127.0.0.0/8` addresses should work out of the box for the loopback device.

**Comment by ialidzhikov** [comment]
/unhold

@plkokanov shared that he wants to be the context around this and wants to provide feedback. Hence, let's wait for his feedback.

**Review by vpnachev — COMMENTED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
/hold
for https://github.com/gardener/gardener/pull/7528#issuecomment-1438165196

**Comment by plkokanov** [comment]
When using `make kind2-up KIND_ENV=local`  as part of the `development/getting_started_locally.md` documentation, the `sudo ip addr add 127.0.0.2 dev lo0` command is necessary to expose the istio ingress gateway of the second seed on `127.0.0.2:443` as `127.0.0.1:443` is already taken by the first seed.

Previously, when only running `make kind2-up` as part of the `deployment/getting_started_locally.md`, which is also executed in prow jobs, the istio ingress gateway of the second seed was exposed on `127.0.0.1:9443` as there was no `ip` command in the prow container and we didn't know about https://github.com/gardener/gardener/pull/7528#issuecomment-1438119297. After recent development, the second seed that is created with `make kind2-up` is also configured to expose the kube-apiserver on the `127.0.0.2` IP, however still on the `9443` port. 

So I actually have two suggestions for improvement:
1. We could expose the istio ingress gateway on `127.0.0.2:443` in all scenarios and get rid of https://github.com/gardener/gardener/blob/16a28f9ae567b5233b786dd72f28466414744183/example/gardener-local/kind/cluster/templates/_extra_port_mappings.tpl#L15-L18
2. Instead of having to add the `127.0.0.2` address manually and documenting it, the `kind-up.sh` script could be improved to do it similar to what it is already doing for the additional IPs required for the multi zonal setup:
https://github.com/gardener/gardener/blob/16a28f9ae567b5233b786dd72f28466414744183/hack/kind-up.sh#L46-L70
https://github.com/gardener/gardener/blob/16a28f9ae567b5233b786dd72f28466414744183/hack/kind-up.sh#L78-L80


The 1st point could probably be handled in a separate PR /cc @timuthy

**Comment by timuthy** [comment]
Nice suggestions @plkokanov. Nevertheless, I suggest to get the doc changes in from this PR and create a separate issue with your suggestions so that it can be tackled by anyone who has capacity.

**Comment by plkokanov** [comment]
> Nevertheless, I suggest to get the doc changes in from this PR and create a separate issue with your suggestions so that it can be tackled by anyone who has capacity.

Sounds ok to me.

/lgtm

**Comment by rfranzke** [comment]
In the light of https://github.com/gardener/gardener/pull/7528#issuecomment-1439547380, can we `/unhold` this PR?

**Comment by rfranzke** [comment]
Just docs:
/milestone v1.65

**Pull request merged**